homeassistant/helpers/event.py

```diff
@@ -231,7 +241,7 @@ def state_change_listener(event: Event) -> None:
         return async_track_state_change_event(hass, entity_ids, state_change_listener)
 
     return hass.bus.async_listen(
-        EVENT_STATE_CHANGED, state_change_dispatcher, event_filter=state_change_filter
+        EVENT_STATE_CHANGED, state_change_dispatcher, event_filter=state_change_filter  # type: ignore[arg-type]
```

Comment:
Is it possible to get rid of this ignore?

I tried converted a few and kept hitting it as well

Reply:
for context 

f644307240d9afd3430e9f2ca11592f3ddce3bf6

Reply:
> Is it possible to get rid of this ignore?

Only if we change `async_listen` itself but that will probably cause a lot of issues currently.
It will be (easily) possible with TypeVar defaults though. https://github.com/home-assistant/core/compare/dev...cdce8p:ha-core:type-event

They are just not yet fully supported by mypy.

Reply:
Its probably best to keep it as an ignore for now.

This change makes it rather obvious where we should be using `[]` instead of `.get()` which is nice.

Reply:
If we do this change, I can check again after some time if it's possible. Most of the issues are `Missing type parameters for generic type "Event"` which should disappear on their own once we update more code to use `EventType[...]`.

The main reason for the issue is just the optional `disallow_any_generics` [mypy setting](https://mypy.readthedocs.io/en/stable/config_file.html?highlight=disallow_any_generics#confval-disallow_any_generics).